FirebaseDatabase/Sources/Realtime/FWebSocketConnection.m

```diff
@@ -179,21 +230,61 @@ - (void)send:(NSDictionary *)dictionary {
     // First send the header so the server knows how many segments are
     // forthcoming
     if (dataSegs.count > 1) {
-        [self.webSocket
-            send:[NSString
-                     stringWithFormat:@"%u", (unsigned int)dataSegs.count]];
+        NSString *formattedData =
+            [NSString stringWithFormat:@"%u", (unsigned int)dataSegs.count];
+#if TARGET_OS_WATCH
+        [self.webSocketTask
+                  sendMessage:[[NSURLSessionWebSocketMessage alloc]
+                                  initWithString:formattedData]
+            completionHandler:^(NSError *_Nullable error) {
+              if (error) {
+                  FFWarn(@"I-RDB083017",
+                         @"Error sending initial web socket data: %@.", error);
+                  return;
+              }
+            }];
+#else
+        [self.webSocket send:formattedData];
+#endif // TARGET_OS_WATCH
     }
 
     // Then, actually send the segments.
     for (NSString *segment in dataSegs) {
+#if TARGET_OS_WATCH
+        [self.webSocketTask sendMessage:[[NSURLSessionWebSocketMessage alloc]
+                                            initWithString:segment]
+                      completionHandler:^(NSError *_Nullable error) {
+                        if (error) {
+                            FFWarn(@"I-RDB083018",
+                                   @"Error sending web socket data: %@", error);
+                            return;
+                        }
+                      }];
+#else
         [self.webSocket send:segment];
+#endif // TARGET_OS_WATCH
     }
 }
 
 - (void)nop:(NSTimer *)timer {
     if (!isClosed) {
         FFLog(@"I-RDB083004", @"(wsc:%@) nop", self.connectionId);
+#if TARGET_OS_WATCH
+        // Note: There's a built in `sendPingWithPongReceiveHandler` call that's
+        // available, but it doesn't match with what the backend is expecting.
+        // Send a string with "0" instead.
```

Comment:
Looks like this comment applies to the non-watch code also?

Reply:
The `sendPingWithPongReceiveHandler` is a `NSURLSessionWebSocketTask` API that's not available with SocketRocket. The "0" being expected from the backend is the same, but I wanted to clarify that this isn't a standard ping/pong that we can use Foundation APIs for.